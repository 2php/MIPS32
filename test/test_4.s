.set noreorder
#Write com function test
start:
  .text 0
  la $s0, 0xbfd003f8
  la $s1, 0xbfd003fc
  li $a0, 0x23
  jal write_com
  nop
  li $a0, 0x24
  jal write_com
  nop
  li $a0, 0x25
  jal write_com
  nop
  break

write_com:
  li $t1, 1
  loop:
  lw $t0, 0($s1)
  andi $t0, $t0, 1
  bne $t0, $t1, loop
  sw $a0, 0($s0)
  jr $ra
