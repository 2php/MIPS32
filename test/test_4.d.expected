booting

fetch instr @ 0x80000000 (10000000000000000000000000000000)
MMU[0x80000000] :  0x00000000
Mem[0x00000000] :  0x3c10bfd0 (00111100000100001011111111010000)
001111 | 00000 | 10000 | 1011111111010000
LUI      rs: 0   rt:16   imm:0xbfd0
R[16] <= 0xbfd00000 (10111111110100000000000000000000)

fetch instr @ 0x80000004 (10000000000000000000000000000100)
MMU[0x80000004] :  0x00000004
Mem[0x00000004] :  0x361003f8 (00110110000100000000001111111000)
001101 | 10000 | 10000 | 0000001111111000
ORI      rs:16   rt:16   imm:0x03f8
R[16] :  0xbfd00000 (10111111110100000000000000000000)
alu_a <= 0xbfd00000 (10111111110100000000000000000000)
alu_b <= 0x000003f8 (00000000000000000000001111111000)
alu_r :  0xbfd003f8 (10111111110100000000001111111000)
R[16] <= 0xbfd003f8 (10111111110100000000001111111000)

fetch instr @ 0x80000008 (10000000000000000000000000001000)
MMU[0x80000008] :  0x00000008
Mem[0x00000008] :  0x3c11bfd0 (00111100000100011011111111010000)
001111 | 00000 | 10001 | 1011111111010000
LUI      rs: 0   rt:17   imm:0xbfd0
R[17] <= 0xbfd00000 (10111111110100000000000000000000)

fetch instr @ 0x8000000c (10000000000000000000000000001100)
MMU[0x8000000c] :  0x0000000c
Mem[0x0000000c] :  0x363103fc (00110110001100010000001111111100)
001101 | 10001 | 10001 | 0000001111111100
ORI      rs:17   rt:17   imm:0x03fc
R[17] :  0xbfd00000 (10111111110100000000000000000000)
alu_a <= 0xbfd00000 (10111111110100000000000000000000)
alu_b <= 0x000003fc (00000000000000000000001111111100)
alu_r :  0xbfd003fc (10111111110100000000001111111100)
R[17] <= 0xbfd003fc (10111111110100000000001111111100)

fetch instr @ 0x80000010 (10000000000000000000000000010000)
MMU[0x80000010] :  0x00000010
Mem[0x00000010] :  0x24040023 (00100100000001000000000000100011)
001001 | 00000 | 00100 | 0000000000100011
ADDIU    rs: 0   rt: 4   imm:0x0023
R[ 0] :  0x00000000 (00000000000000000000000000000000)
alu_a <= 0x00000000 (00000000000000000000000000000000)
alu_b <= 0x00000023 (00000000000000000000000000100011)
alu_r :  0x00000023 (00000000000000000000000000100011)
R[ 4] <= 0x00000023 (00000000000000000000000000100011)

fetch instr @ 0x80000014 (10000000000000000000000000010100)
MMU[0x80000014] :  0x00000014
Mem[0x00000014] :  0x0c00000e (00001100000000000000000000001110)
000011 | 00000000000000000000001110
JAL      instr_index:0x000000e
R[31] <= 0x8000001c (10000000000000000000000000011100)

fetch instr @ 0x80000038 (10000000000000000000000000111000)
MMU[0x80000038] :  0x00000038
Mem[0x00000038] :  0x24090001 (00100100000010010000000000000001)
001001 | 00000 | 01001 | 0000000000000001
ADDIU    rs: 0   rt: 9   imm:0x0001
R[ 0] :  0x00000000 (00000000000000000000000000000000)
alu_a <= 0x00000000 (00000000000000000000000000000000)
alu_b <= 0x00000001 (00000000000000000000000000000001)
alu_r :  0x00000001 (00000000000000000000000000000001)
R[ 9] <= 0x00000001 (00000000000000000000000000000001)

fetch instr @ 0x8000003c (10000000000000000000000000111100)
MMU[0x8000003c] :  0x0000003c
Mem[0x0000003c] :  0x8e280000 (10001110001010000000000000000000)
100011 | 10001 | 01000 | 0000000000000000
LW       rs:17   rt: 8   imm:0x0000
R[17] :  0xbfd003fc (10111111110100000000001111111100)
alu_a <= 0xbfd003fc (10111111110100000000001111111100)
alu_b <= 0x00000000 (00000000000000000000000000000000)
alu_r :  0xbfd003fc (10111111110100000000001111111100)
MMU[0xbfd003fc] :  0x1fd003fc
Mem[0x1fd003fc] :  0x00000001 (00000000000000000000000000000001)
R[ 8] <= 0x00000001 (00000000000000000000000000000001)

fetch instr @ 0x80000040 (10000000000000000000000001000000)
MMU[0x80000040] :  0x00000040
Mem[0x00000040] :  0x31080001 (00110001000010000000000000000001)
001100 | 01000 | 01000 | 0000000000000001
ANDI     rs: 8   rt: 8   imm:0x0001
R[ 8] :  0x00000001 (00000000000000000000000000000001)
alu_a <= 0x00000001 (00000000000000000000000000000001)
alu_b <= 0x00000001 (00000000000000000000000000000001)
alu_r :  0x00000001 (00000000000000000000000000000001)
R[ 8] <= 0x00000001 (00000000000000000000000000000001)

fetch instr @ 0x80000044 (10000000000000000000000001000100)
MMU[0x80000044] :  0x00000044
Mem[0x00000044] :  0x1509fffd (00010101000010011111111111111101)
000101 | 01000 | 01001 | 1111111111111101
BNE      rs: 8   rt: 9   imm:0xfffd
R[ 8] :  0x00000001 (00000000000000000000000000000001)
R[ 9] :  0x00000001 (00000000000000000000000000000001)

fetch instr @ 0x80000048 (10000000000000000000000001001000)
MMU[0x80000048] :  0x00000048
Mem[0x00000048] :  0xae040000 (10101110000001000000000000000000)
101011 | 10000 | 00100 | 0000000000000000
SW       rs:16   rt: 4   imm:0x0000
R[16] :  0xbfd003f8 (10111111110100000000001111111000)
R[ 4] :  0x00000023 (00000000000000000000000000100011)
alu_a <= 0xbfd003f8 (10111111110100000000001111111000)
alu_b <= 0x00000000 (00000000000000000000000000000000)
alu_r :  0xbfd003f8 (10111111110100000000001111111000)
MMU[0xbfd003f8] :  0x1fd003f8
Mem[0x1fd003f8] <= 0x23 (00100011)

fetch instr @ 0x8000004c (10000000000000000000000001001100)
MMU[0x8000004c] :  0x0000004c
Mem[0x0000004c] :  0x03e00008 (00000011111000000000000000001000)
000000 | 11111 | 00000 | 00000 | 00000 | 001000
SPECIAL  rs:31   rt: 0   rd: 0   sa: 0   JR     
R[31] :  0x8000001c (10000000000000000000000000011100)

fetch instr @ 0x8000001c (10000000000000000000000000011100)
MMU[0x8000001c] :  0x0000001c
Mem[0x0000001c] :  0x24040024 (00100100000001000000000000100100)
001001 | 00000 | 00100 | 0000000000100100
ADDIU    rs: 0   rt: 4   imm:0x0024
R[ 0] :  0x00000000 (00000000000000000000000000000000)
alu_a <= 0x00000000 (00000000000000000000000000000000)
alu_b <= 0x00000024 (00000000000000000000000000100100)
alu_r :  0x00000024 (00000000000000000000000000100100)
R[ 4] <= 0x00000024 (00000000000000000000000000100100)

fetch instr @ 0x80000020 (10000000000000000000000000100000)
MMU[0x80000020] :  0x00000020
Mem[0x00000020] :  0x0c00000e (00001100000000000000000000001110)
000011 | 00000000000000000000001110
JAL      instr_index:0x000000e
R[31] <= 0x80000028 (10000000000000000000000000101000)

fetch instr @ 0x80000038 (10000000000000000000000000111000)
MMU[0x80000038] :  0x00000038
Mem[0x00000038] :  0x24090001 (00100100000010010000000000000001)
001001 | 00000 | 01001 | 0000000000000001
ADDIU    rs: 0   rt: 9   imm:0x0001
R[ 0] :  0x00000000 (00000000000000000000000000000000)
alu_a <= 0x00000000 (00000000000000000000000000000000)
alu_b <= 0x00000001 (00000000000000000000000000000001)
alu_r :  0x00000001 (00000000000000000000000000000001)
R[ 9] <= 0x00000001 (00000000000000000000000000000001)

fetch instr @ 0x8000003c (10000000000000000000000000111100)
MMU[0x8000003c] :  0x0000003c
Mem[0x0000003c] :  0x8e280000 (10001110001010000000000000000000)
100011 | 10001 | 01000 | 0000000000000000
LW       rs:17   rt: 8   imm:0x0000
R[17] :  0xbfd003fc (10111111110100000000001111111100)
alu_a <= 0xbfd003fc (10111111110100000000001111111100)
alu_b <= 0x00000000 (00000000000000000000000000000000)
alu_r :  0xbfd003fc (10111111110100000000001111111100)
MMU[0xbfd003fc] :  0x1fd003fc
Mem[0x1fd003fc] :  0x00000001 (00000000000000000000000000000001)
R[ 8] <= 0x00000001 (00000000000000000000000000000001)

fetch instr @ 0x80000040 (10000000000000000000000001000000)
MMU[0x80000040] :  0x00000040
Mem[0x00000040] :  0x31080001 (00110001000010000000000000000001)
001100 | 01000 | 01000 | 0000000000000001
ANDI     rs: 8   rt: 8   imm:0x0001
R[ 8] :  0x00000001 (00000000000000000000000000000001)
alu_a <= 0x00000001 (00000000000000000000000000000001)
alu_b <= 0x00000001 (00000000000000000000000000000001)
alu_r :  0x00000001 (00000000000000000000000000000001)
R[ 8] <= 0x00000001 (00000000000000000000000000000001)

fetch instr @ 0x80000044 (10000000000000000000000001000100)
MMU[0x80000044] :  0x00000044
Mem[0x00000044] :  0x1509fffd (00010101000010011111111111111101)
000101 | 01000 | 01001 | 1111111111111101
BNE      rs: 8   rt: 9   imm:0xfffd
R[ 8] :  0x00000001 (00000000000000000000000000000001)
R[ 9] :  0x00000001 (00000000000000000000000000000001)

fetch instr @ 0x80000048 (10000000000000000000000001001000)
MMU[0x80000048] :  0x00000048
Mem[0x00000048] :  0xae040000 (10101110000001000000000000000000)
101011 | 10000 | 00100 | 0000000000000000
SW       rs:16   rt: 4   imm:0x0000
R[16] :  0xbfd003f8 (10111111110100000000001111111000)
R[ 4] :  0x00000024 (00000000000000000000000000100100)
alu_a <= 0xbfd003f8 (10111111110100000000001111111000)
alu_b <= 0x00000000 (00000000000000000000000000000000)
alu_r :  0xbfd003f8 (10111111110100000000001111111000)
MMU[0xbfd003f8] :  0x1fd003f8
Mem[0x1fd003f8] <= 0x24 (00100100)

fetch instr @ 0x8000004c (10000000000000000000000001001100)
MMU[0x8000004c] :  0x0000004c
Mem[0x0000004c] :  0x03e00008 (00000011111000000000000000001000)
000000 | 11111 | 00000 | 00000 | 00000 | 001000
SPECIAL  rs:31   rt: 0   rd: 0   sa: 0   JR     
R[31] :  0x80000028 (10000000000000000000000000101000)

fetch instr @ 0x80000028 (10000000000000000000000000101000)
MMU[0x80000028] :  0x00000028
Mem[0x00000028] :  0x24040025 (00100100000001000000000000100101)
001001 | 00000 | 00100 | 0000000000100101
ADDIU    rs: 0   rt: 4   imm:0x0025
R[ 0] :  0x00000000 (00000000000000000000000000000000)
alu_a <= 0x00000000 (00000000000000000000000000000000)
alu_b <= 0x00000025 (00000000000000000000000000100101)
alu_r :  0x00000025 (00000000000000000000000000100101)
R[ 4] <= 0x00000025 (00000000000000000000000000100101)

fetch instr @ 0x8000002c (10000000000000000000000000101100)
MMU[0x8000002c] :  0x0000002c
Mem[0x0000002c] :  0x0c00000e (00001100000000000000000000001110)
000011 | 00000000000000000000001110
JAL      instr_index:0x000000e
R[31] <= 0x80000034 (10000000000000000000000000110100)

fetch instr @ 0x80000038 (10000000000000000000000000111000)
MMU[0x80000038] :  0x00000038
Mem[0x00000038] :  0x24090001 (00100100000010010000000000000001)
001001 | 00000 | 01001 | 0000000000000001
ADDIU    rs: 0   rt: 9   imm:0x0001
R[ 0] :  0x00000000 (00000000000000000000000000000000)
alu_a <= 0x00000000 (00000000000000000000000000000000)
alu_b <= 0x00000001 (00000000000000000000000000000001)
alu_r :  0x00000001 (00000000000000000000000000000001)
R[ 9] <= 0x00000001 (00000000000000000000000000000001)

fetch instr @ 0x8000003c (10000000000000000000000000111100)
MMU[0x8000003c] :  0x0000003c
Mem[0x0000003c] :  0x8e280000 (10001110001010000000000000000000)
100011 | 10001 | 01000 | 0000000000000000
LW       rs:17   rt: 8   imm:0x0000
R[17] :  0xbfd003fc (10111111110100000000001111111100)
alu_a <= 0xbfd003fc (10111111110100000000001111111100)
alu_b <= 0x00000000 (00000000000000000000000000000000)
alu_r :  0xbfd003fc (10111111110100000000001111111100)
MMU[0xbfd003fc] :  0x1fd003fc
Mem[0x1fd003fc] :  0x00000001 (00000000000000000000000000000001)
R[ 8] <= 0x00000001 (00000000000000000000000000000001)

fetch instr @ 0x80000040 (10000000000000000000000001000000)
MMU[0x80000040] :  0x00000040
Mem[0x00000040] :  0x31080001 (00110001000010000000000000000001)
001100 | 01000 | 01000 | 0000000000000001
ANDI     rs: 8   rt: 8   imm:0x0001
R[ 8] :  0x00000001 (00000000000000000000000000000001)
alu_a <= 0x00000001 (00000000000000000000000000000001)
alu_b <= 0x00000001 (00000000000000000000000000000001)
alu_r :  0x00000001 (00000000000000000000000000000001)
R[ 8] <= 0x00000001 (00000000000000000000000000000001)

fetch instr @ 0x80000044 (10000000000000000000000001000100)
MMU[0x80000044] :  0x00000044
Mem[0x00000044] :  0x1509fffd (00010101000010011111111111111101)
000101 | 01000 | 01001 | 1111111111111101
BNE      rs: 8   rt: 9   imm:0xfffd
R[ 8] :  0x00000001 (00000000000000000000000000000001)
R[ 9] :  0x00000001 (00000000000000000000000000000001)

fetch instr @ 0x80000048 (10000000000000000000000001001000)
MMU[0x80000048] :  0x00000048
Mem[0x00000048] :  0xae040000 (10101110000001000000000000000000)
101011 | 10000 | 00100 | 0000000000000000
SW       rs:16   rt: 4   imm:0x0000
R[16] :  0xbfd003f8 (10111111110100000000001111111000)
R[ 4] :  0x00000025 (00000000000000000000000000100101)
alu_a <= 0xbfd003f8 (10111111110100000000001111111000)
alu_b <= 0x00000000 (00000000000000000000000000000000)
alu_r :  0xbfd003f8 (10111111110100000000001111111000)
MMU[0xbfd003f8] :  0x1fd003f8
Mem[0x1fd003f8] <= 0x25 (00100101)

fetch instr @ 0x8000004c (10000000000000000000000001001100)
MMU[0x8000004c] :  0x0000004c
Mem[0x0000004c] :  0x03e00008 (00000011111000000000000000001000)
000000 | 11111 | 00000 | 00000 | 00000 | 001000
SPECIAL  rs:31   rt: 0   rd: 0   sa: 0   JR     
R[31] :  0x80000034 (10000000000000000000000000110100)

fetch instr @ 0x80000034 (10000000000000000000000000110100)
MMU[0x80000034] :  0x00000034
Mem[0x00000034] :  0x0000000c (00000000000000000000000000001100)
000000 | 00000 | 00000 | 00000 | 00000 | 001100
SPECIAL  rs: 0   rt: 0   rd: 0   sa: 0   SYSCALL
halt
